File: bus-booking-frontend/src/services/userService.js
import api from "../api/axiosInstance";

export const getAllUsers  = ()      =>
  api.get("/users");

export const getUserById  = (id)    =>
  api.get(`/users/${id}`);

export const deleteUser   = (id)    =>
  api.delete(`/users/${id}`);

export const updateUser   = (id, d) =>
  api.put(`/users/${id}`, d);

File: BusTicketBookingSystem/src/main/java/com/ash/bbus/web/BusTicketBookingSystem/service/UserService.java
package com.ash.bbus.web.BusTicketBookingSystem.service;

import com.ash.bbus.web.BusTicketBookingSystem.dto.RegisterDTO;
import com.ash.bbus.web.BusTicketBookingSystem.dto.UserResponseDTO;
import com.ash.bbus.web.BusTicketBookingSystem.dto.UserUpdateDTO;

import java.util.List;

public interface UserService {

	UserResponseDTO register(RegisterDTO dto);

	// ✅ New — admin registration
	UserResponseDTO registerAdmin(RegisterDTO dto, String secretKey);

	UserResponseDTO getUserById(Long id);

	UserResponseDTO getUserByEmail(String email);

	List<UserResponseDTO> getAllUsers();

	UserResponseDTO updateUser(Long id, UserUpdateDTO dto);

	void deleteUser(Long id);
}